### PR TITLE
Decompile Remaining `w_015` Functions

### PR DESCRIPTION
**Pull request by hohle**
* `EntityWeaponAttack`
* `func_ptr_80170010`
* `func_ptr_8017000C`

### CODE REVIEW

**Comment by hohle** [comment]
Thanks to @bismurphy for help on `func_ptr_8017000C`! 🎉

File: src/weapon/w_015.c [hunk 1]
```diff
@@ -9,7 +9,97 @@ extern s32 g_HandId;
...
+            self->posX.i.hi = PLAYER.posX.i.hi + PLAYER.hitboxOffX;
+        }
+
+        SetWeaponProperties(self, 0);
+        DestroyEntityWeapon(true);
+
+        self->hitboxWidth = 4;
+        self->hitboxHeight = 4;
+        if (((self->params >> 8) & 0x7F) == 1) {
+            self->drawFlags |= 4;
```
Comment (Xeeynamo) [inline src/weapon/w_015.c:52]:
```suggestion
            self->drawFlags |= DRAW_COLORS;
```
and similar

Comment (bismurphy) [inline src/weapon/w_015.c:52]: Should be FLAG_DRAW_ROTZ I believe

**Review by Xeeynamo — APPROVED** [review]
:ok_hand:

File: src/weapon/w_015.c [hunk 2]
```diff
@@ -9,7 +9,97 @@ extern s32 g_HandId;
...
+            self->unk5A = 0x66;
+        }
+        self->animCurFrame = D_6D000_8017A6B8[((self->params >> 8) & 0x7F)];
+        self->facingLeft = PLAYER.facingLeft;
+        self->zPriority = PLAYER.zPriority - 2;
+        self->flags = FLAG_UNK_08000000;
+
+        SetSpeedX(FIX(8));
+        if (((self->params >> 8) & 0x7F) == 3) {
+            SetSpeedX(FIX(349525.0 / 65536.0));
```
Comment (bismurphy) [inline src/weapon/w_015.c:33]: Does this work if you put it as 5.3333?

File: src/weapon/w_015.c [hunk 3]
```diff
@@ -76,9 +166,314 @@ static void func_ptr_80170008(Entity* self) {
...
+        }
+
+        self->facingLeft = PLAYER.facingLeft;
+        self->zPriority = PLAYER.zPriority - 2;
+        self->flags = FLAG_UNK_08000000 | FLAG_UNK_800000;
+
+        self->primIndex = g_api.AllocPrimitives(PRIM_TILE, 32);
+
+        if (self->primIndex == -1) {
+            self->flags &= ~FLAG_UNK_800000;
```
Comment (bismurphy) [inline src/weapon/w_015.c:200]: I'm not sure why FLAG_UNK_800000 still exists, this should be FLAG_HAS_PRIMS

File: src/weapon/w_015.c [hunk 4]
```diff
@@ -76,9 +166,314 @@ static void func_ptr_80170008(Entity* self) {
...
+                self, ((g_HandId + 1) << 0xC) + FACTORY(0x1000, 78), 0);
+            g_api.func_80102CD8(3);
+            self->step++;
+            self->velocityY = FIX(-6);
+            self->ext.weapon.lifetime = 16;
+
+            if (self->flags & FLAG_UNK_800000) {
+                prim = &g_PrimBuf[self->primIndex];
+                while (prim != NULL) {
+                    prim->drawMode |= 8;
```
Comment (bismurphy) [inline src/weapon/w_015.c:310]: DRAW_HIDE

File: src/weapon/w_015.c [hunk 5]
```diff
@@ -76,9 +166,314 @@ static void func_ptr_80170008(Entity* self) {
...
+        break;
+
+    case 3:
+        self->hitboxState = 0;
+        // the last three bits of weapon lifetime need to be zero.
+        if ((self->ext.weapon.lifetime & 7) == 0) {
+            g_api.PlaySfx(SFX_FM_EXPLODE_B);
+        }
+        // no, seriously. the last three bits of weapon lifetime really need
+        // to be zero. and don't trust that last comparison. i'm not joking.
```
Comment (bismurphy) [inline src/weapon/w_015.c:379]: this comment is great

File: src/weapon/w_015.c [hunk 6]
```diff
@@ -76,9 +166,314 @@ static void func_ptr_80170008(Entity* self) {
...
+            return;
+        }
+
+        break;
+    }
+}
+
+extern SpriteParts D_6D000_8017A2B0[];
+extern AnimationFrame D_6D000_8017A770[];
+extern s16 D_6D000_8017A78C[];
```
Comment (bismurphy) [inline src/weapon/w_015.c:397]: Probably unify the externs at the top of the file

File: src/weapon/w_015.c [hunk 7]
```diff
@@ -76,9 +166,314 @@ static void func_ptr_80170008(Entity* self) {
...
+
+s32 func_ptr_80170010(Entity* self) {
+    s16 angle;
+    s32 accesserationX;
+    s32 accelerationY;
```
Comment (bismurphy) [inline src/weapon/w_015.c:402]: typo on the X value; and maybe just shorten these to accelX and accelY? Just makes them a bit more readable

Comment (hohle) [reply src/weapon/w_015.c:399]: I fixed the typo but left the name since they match the field names.

File: src/weapon/w_015.c (hunk 6 above)
Comment (hohle) [reply src/weapon/w_015.c]: Moved.

File: src/weapon/w_015.c (hunk 3 above)
Comment (hohle) [reply src/weapon/w_015.c]: I fixed this one and will submit another PR removing the `_UNK_` flag.

**Pull request merged**